File: 1. Study/0527 - 완전탐색(블루투포스)/mojaeya/4_졸업선물.js
let arr = [
  [6, 6],
  [2, 2],
  [4, 3],
  [4, 5],
  [10, 3],
];

console.log(solution(5, 28, arr));

function solution(N, M, goods) {
  let answer = 0;
  let count = 0;
  let temp = [];
  goods.sort((a, b) => a[0] + a[1] - (b[0] + b[1]));

  for (let i = 0; i < N; i++) {
    let budget = M - (goods[i][0] / 2 + goods[i][1]);
    count = 1;
    for (let j = 0; j < N; j++) {
      if (j !== i && goods[j][0] + goods[j][1] <= budget) {
        budget -= goods[j][0] + goods[j][1];
        count++;
      } else break;
    }
    answer = Math.max(answer, count);
  }
  return answer;
}

File: 1. Study/0527 - 완전탐색(블루투포스)/kimshin/4.졸업선물.py
def sol(m,product):
    answer=0
    n=len(product)
    product.sort(key=lambda x:(x[0]+x[1]))
    for i in range(n):
        money = m-(product[i][0] // 2 + product[i][1])
        cnt=1
        for j in range(n):
            if j != i and product[j][0] + product[j][1] > money:
                break
            if j != i and product[j][0] + product[j][1] <= money:
                money -= product[j][0] + product[j][1]
                cnt+=1
        answer = max(answer,cnt)
    return answer



arr=[
    [6,6],
    [2,2],
    [4,3],
    [4,5],
    [10,3]
]
print(sol(28,arr))

# https://soraji.github.io/code/2021/04/19/JS33/